Add ETDump event tracer support to LLaMa runner

navsud opened the pull request:
Summary:
for op-level profiling of od-llms

**Addresses reviewer feedback:**
- Added `ET_EVENT_TRACER_ENABLED` ifdef guards around ETDump code to ensure normal builds compile without event tracer support
- Replaced hardcoded `/data/local/tmp/etdump.bin` path with configurable `--etdump_path` flag (default: `"etdump.in"`)
- ETDumpGen is created only when compiled with event tracer support enabled

Differential Revision: D87122487

kimishpatel reviewed (approved):
Review automatically exported from Phabricator review in Meta.

navsud committed:
Add ETDump event tracer support to LLaMa runner (#15934)

Summary:

for op-level profiling of od-llms

**Addresses reviewer feedback:**
- Added `ET_EVENT_TRACER_ENABLED` ifdef guards around ETDump code to ensure normal builds compile without event tracer support
- Replaced hardcoded `/data/local/tmp/etdump.bin` path with configurable `--etdump_path` flag (default: `"etdump.in"`)
- ETDumpGen is created only when compiled with event tracer support enabled

Reviewed By: larryliu0820, kimishpatel

Differential Revision: D87122487
